core/src/consensus.rs

```diff
@@ -533,6 +564,14 @@ impl Tower {
         bank_weights.pop().map(|b| b.2)
     }
 
+    pub fn adjust_lockouts_if_newer_root(&mut self, root_slot: Slot) {
+        let my_root_slot = self.lockouts.root_slot.unwrap_or(0);
+        if root_slot > my_root_slot {
+            self.lockouts.root_slot = Some(root_slot);
+            self.lockouts.votes.retain(|v| v.slot > root_slot);
```

Comment:
@t-nelson I think this part is why we need to consult the ancestors, what if the votes higher than the `root_slot` don't descend from `root_slot` here